device/receive.go

```diff
@@ -103,100 +117,117 @@ func (device *Device) RoutineReceiveIncoming(recv conn.ReceiveFunc) {
 			if deathSpiral < 10 {
 				deathSpiral++
 				time.Sleep(time.Second / 3)
-				buffer = device.GetMessageBuffer()
 				continue
 			}
 			return
 		}
 		deathSpiral = 0
 
-		if size < MinMessageSize {
-			continue
-		}
+		// handle each packet in the batch
+		for i, size := range sizes[:count] {
+			if size < MinMessageSize {
+				// this is also deliberately skipping len 0
```

Comment:
Not sure I understand this comment. I mean, I get that size < X where X > 0 means that it skips len 0. But what's the significance of a 0 packet? Why should that be pointed out more often than a size 7 or size 4 packet? They're all invalid, right? Or is there some interesting circumstance in the new recvmmsg code that yields size 0 packets that I should be aware of?

Reply:
This came from usage in tailscaled's `conn.Bind` implementation where len 0 is expected when we intentionally handle/swallow something that was part of a batch. It's not relevant here. I've removed the comment in https://github.com/WireGuard/wireguard-go/pull/64/commits/933c8b166dd41ddeb5d87ea505624a2fa34848ae.